core/src/main/java/org/infinispan/interceptors/impl/NotificationInterceptor.java

```diff
@@ -18,6 +18,14 @@
  */
 public class NotificationInterceptor extends DDSequentialInterceptor {
    private CacheNotifier notifier;
+   private ReturnHandler transactionCompleteReturnHandler = (ctx1, command1, rv, throwable) -> {
+      if (throwable != null)
+         return null;
```

Comment:
what is the difference between throwing the ```throwable``` and return ```null```? They do the same IMO but in some places you throw it and in others you just return ```null```.
I'm wondering if would it be useful to create 2 new interfaces/methods. One to be invoked on a successful execution and the other in a failed execution. Just because I'm seeing a lot of this checks.